src/data.js

```diff
@@ -143,54 +143,56 @@ Namespace.ensureNamespace = function(namespace, expires) {
 /**Delete expired entries */
 Namespace.expireEntries = async function(now) {
   let continuationToken = undefined;
+  let count = 0;
+
   while (1) {
     let data = await this.scan({
       expires: Entity.op.lessThan(now),
     },
     {
-      limit:         500,
+      limit:         100,
       continuation:   continuationToken,
     });
-    var dataLength = data.entries.length;
-    
-    for (var i=0; i<dataLength; i++) {
-      let entry = data.entries[i];
-      let namespace = entry.parent + '.' + entry.name;
-      if (entry.parent.length === 0 || entry.name.length === 0) {
-        namespace = entry.parent + entry.name;
-      }
+
+    await Promise.all(data.entries.map(entry => {
       // insertTask is careful to update expires of entries 
       // from the root out to the leaf. A parent's expires is
       // always later than the child's. Hence, we can delete an
       // entry without checking its children.
       entry.remove(false, true);
```

Comment:
You are not returning the promise... so you'll have a LOT of requests in the air...

Upside is that you keep scanning without waiting for things to be deleted... downside is that it might run out of ports... who knows :)

Reply:
Or run out of memory, because I think the agent we use by default is limited to like 50 connections or so...

Reply:
Yikes, Rust has ruined me!  You're right :)